chef/cookbooks/crowbar/templates/default/crowbar-pre-upgrade.sh.erb

```diff
@@ -25,6 +25,13 @@ if [[ -f /usr/bin/neutron-ha-tool ]] ; then
     OS_AUTH_URL="<%= @os_auth_url_v2 %>" neutron-ha-tool --l3-agent-evacuate $(hostname)
 fi
 
+# Even if agents are migrated, we want to explicitly destroy the router namespaces
+# because in some upgrade cases, the routers remain active when there are no
+# valid destinations for the migration
```

Comment:
@rhafer first note that this PR may be superseded by https://github.com/crowbar/crowbar-core/pull/779

These two PRs address the same problem. In a three controller scenario, when you want to upgrade the first one, the routers are migrated to some other controller. This works flawlessly. Let's say we just upgraded first controller (node1) and routers migrated to the third controller (node3). So, right after node1 is upgraded, chef-clients runs there and Neutron and the agents are created. It happens then that node1 has a router but so does node3. This is a conflicting scenario that often leads to VM connectivity loss.

So the solution is to mercilessly kill the routers in the other controllers once the first one is upgraded. This is what this PR aimed to to, but I think it's better addresses in #779 . There's going to be for sure an interval where the two routers will co-exist so we expect misbehavior for some (hopefully short) time.